wgsl/index.bs

```diff
@@ -5690,6 +5690,359 @@ Note: If the function [=return value|returns a value=], that value is ignored.
+        *s1*: *B1*<br>
+        *s2*: *B2*
+    <td class="nowrap">*B*∖{Default} &cup; B1 &cup; B2
+  <tr>
+    <td class="nowrap">loop {*s*}
+    <td class="nowrap">*s*: *B*
+    <td class="nowrap">let b = *B*∖{Default};<br>
+        if (Break in b) b = b∖{Break} &cup; {Default};<br>
+        b∖{Continue}
+  <tr>
+    <td class="nowrap">loop {*s1* continuing {*s2*}}
+    <td class="nowrap">*s1*: *B1*<br>
+        *s2*: *B2*<br>
+        Neither Continue nor Return are in *B2*
+    <td class="nowrap">let b = (*B1* &cup; *B2*)∖{Default};<br>
+        if (Break in b) b = b∖{Break} &cup; {Default};<br>
+        b∖{Continue}
+  <tr>
+    <td class="nowrap">switch(*e*) {case _: *s_1* ... case _: *s_n*}
+    <td class="nowrap">*e*: *B*<br>
+        *s_1*: *B_1*<br>
+        ...<br>
+        *s_n*: *B_n*<br>
+        Fallthrough is not in *B_n*
+    <td class="nowrap">let b = *B*∖{Default} &cup; B_1 &cup; ... &cup; B_n;<br>
+        if (Break in b) b = b∖{Break} &cup; {Default};<br>
+        b∖{Fallthrough}
+</table>
+
+for loops get desugared before this analysis is performed, see [[#for-statement]].
+Similarly, if statements without an else branch are treated as if they had an empty else branch (which adds Default to their behavior).
+
+<table class='data'>
+  <caption>Rules for analyzing and validating the behaviors of expressions</caption>
+  <thead>
+    <tr><th>Expression<th>Preconditions<th>Resulting behavior
+  </thead>
+  <tr>
+    <td class="nowrap">f(*e_1*, ..., *e_n*)
+    <td class="nowrap">*e_1*: *B_1*<br>
+        ...<br>
+        *e_n*: *B_n*<br>
+        *f*'s body has behavior *B*
+    <td class="nowrap">*B* &cup; (*B_1* &cup; ... &cup; *B_n*)∖{Default}
+  <tr>
+    <td class="nowrap">Any literal
+    <td>
+    <td class="nowrap">{Default}
+  <tr>
+    <td class="nowrap">Any variable reference
+    <td>
+    <td class="nowrap">{Default}
+  <tr>
+    <td class="nowrap">e1[e2]
+    <td class="nowrap">*e1*: *B1*<br>
+        *e2*: *B2*
+    <td class="nowrap">*B1* &cup; *B2*
+  <tr>
+    <td class="nowrap">e.field
+    <td class="nowrap">*e*: *B*
+    <td class="nowrap">*B*
+  <tr>
+    <td class="nowrap">e1 || e2
+    <td class="nowrap">*e1*: *B1*<br>
+        *e2*: *B2*
+    <td class="nowrap">*B1* &cup; *B2*
+  <tr>
+    <td class="nowrap">e1 && e2
+    <td class="nowrap">*e1*: *B1*<br>
+        *e2*: *B2*
+    <td class="nowrap">*B1* &cup; *B2*
+</table>
+
+All functions in the standard library count as having a body with behavior {Default}.
+
+The shader must be rejected if it violates the rules for functions (above) or if it contains a statement or expression where none of the rules above apply because of preconditions.
+
+### Notes ### {#behaviors-notes}
+
+Here is the full list of ways that these rules can cause a program to be rejected (this is just restating information already listed above):
+- The body of a function (treated as a regular statement) has a behavior not included in {Default, Return, Discard}.
```

Comment:
We can fix this after landing, but FYI:
    

           [=statement/fallthrough=]

is a link to the definition.  The 'statement' part is there because the original definition is contextualized with a dfn-for attribute:


              <dfn noexport dfn-for="statement">fallthrough</dfn>